### PR TITLE
make screenshot sound use touch sounds setting

### PR DESCRIPTION
**Pull request by x86pup**
Removes screenshot sound instead of adding a toggle

### CODE REVIEW

**Comment by x86pup** [comment]
Solves https://github.com/GrapheneOS/os-issue-tracker/issues/1696

**Comment by muhomorr** [comment]
Can the screenshot capture sound be removed entirely instead?

**Comment by thestinger** [comment]
That's an option but I'm not sure what is best.

**Comment by muhomorr** [comment]
There's a prominent visual indication after screenshot is captured.
If TalkBack is enabled, there's a "Saving screenshot" announcement.

**Comment by x86pup** [comment]
I can look into removing it. Do you want just the screenshot sound method call removed like this? Or do you want the entirety of it removed? The latter will make it harder to port in the future though and not sure if it will break anything else.

**Comment by muhomorr** [comment]
There's one more caller of `playCameraSound()`.
Simpler to just remove contents of `playCameraSound()`.

**Comment by x86pup** [comment]
Do we also want to remove the visual indication of a screenshot?

**Comment by muhomorr** [comment]
I think not. I meant that the sound is redundant, since there's both a visual indication and a TalkBack announcement.

**Comment by x86pup** [comment]
Removed screenshot sound instead, tested on my Pixel 5 screenshot sound is indeed gone.

**Comment by muhomorr** [comment]
I think better to remove `playCameraSound()` method too, it'll make sure build will fail in case there's more callers of it in the future.
Alternatively, can just remove contents of `playCameraSound()`.

**Comment by x86pup** [comment]
@muhomorr Removed the contents of playCameraSound() instead

**Review by muhomorr — APPROVED** [review]

File: packages/SystemUI/src/com/android/systemui/screenshot/ScreenshotController.java [hunk 1]
```diff
@@ -854,8 +854,11 @@ private void playCameraSound() {
      * failure).
      */
     private void saveScreenshotAndToast(Consumer<Uri> finisher) {
-        // Play the shutter sound to notify that we've taken a screenshot
-        playCameraSound();
+        if (Settings.System.getInt(mContext.getContentResolver(),
```
Comment (muhomorr) [inline packages/SystemUI/src/com/android/systemui/screenshot/ScreenshotController.java]: Better to move the check inside `playCameraSound()`

Comment (x86pup) [reply packages/SystemUI/src/com/android/systemui/screenshot/ScreenshotController.java]: Moved the check

File: packages/SystemUI/src/com/android/systemui/screenshot/ScreenshotController.java [hunk 2]
```diff
@@ -838,15 +838,18 @@ private ListenableFuture<MediaPlayer> loadCameraSound() {
     }
 
     private void playCameraSound() {
-        mCameraSound.addListener(() -> {
-            try {
-                MediaPlayer player = mCameraSound.get();
-                if (player != null) {
-                    player.start();
+        if (Settings.System.getInt(mContext.getContentResolver(),
```
Comment (muhomorr) [inline packages/SystemUI/src/com/android/systemui/screenshot/ScreenshotController.java]: Better to return early instead, it'll likely simplify porting.

Comment (x86pup) [reply packages/SystemUI/src/com/android/systemui/screenshot/ScreenshotController.java]: Made it return early if the setting is not true/enabled

**Pull request merged**